src/pages/FrontPage/FrontPage.jsx

```diff
@@ -0,0 +1,114 @@
+import styling from './FrontPage.module.css';
+import panelImg from '../../assets/images/panelImage.jpg';
+import pencil from '../../assets/images/brokenPencil.png';
+import { useState, useEffect } from 'react';
+import axios from 'axios';
+
+const baseUrl = import.meta.env.VITE_API_BASE_URL;
+
+const FrontPage = () => {
+  const [courseDescriptions, setCoursesDescriptions] = useState([]);
+  const [state, setState] = useState({
+    loading: true,
+    error: false,
+  });
+
+  useEffect(() => {
+    axios
+      .get(`${baseUrl}/course/public`)
```

Comment:
The main problem I found when I didn't have the right URL in my .env file was that the error wasn't getting handled properly because the FrontPage checks `data.success === false`, which wasn't getting caught when the success property doesn't exist:
<img width="241" alt="image" src="https://github.com/user-attachments/assets/bf72ae65-48d3-4c8a-a15a-1ffd4aae3db7">

So I started by adding the success property into the successful response object, but since we aren't using the error message it made more sense to just return the data or nothing instead. Now this function is just calling axios in one line, and at that point it's much more concise to handle the promise here than have a try catch that creates lots of indentation.

In the end, this would have been fine as is. I think this is more concise and less branching logic.